lib/facter/ethtool.rb

```diff
@@ -15,47 +15,70 @@ def ethtool(args)
   def ethtool_result_to_h(cmd_output, has_no_headlines = false)
     result = {}
     current_subsection = result
+    current_list = nil
 
     lines = cmd_output.split("\n")
-    lines = has_no_headlines ? lines : lines.drop(1) # omit first line if it is a headline
+    lines = has_no_headlines ? lines : lines.drop(1)
+
     lines.each do |line|
       line = line.strip
+      next if line.empty?
+
       if line.end_with?(':')
-        if has_no_headlines # case 1: key with empty value
+        current_list = nil
+        if has_no_headlines
           current_subsection[line.tr(':', '')] = ''
-        else # case 2: headline which becomes key to a sub-hash
+        else
           subsection_title = line.chop
           subsection_title = subsection_title.tr(' ', '_')
           result[subsection_title] = {}
           current_subsection = result[subsection_title]
         end
-      else # case 3: normal key-value pair
+      elsif %r{:\s+}.match?(line)
+        current_list = nil
         parts = line.split(%r{:\s+})
         if parts.length == 2
-          current_subsection[parts[0].strip.tr(' ', '_')] = parts[1].strip
+          # Move PTP Hardware Clock to root level
+          if parts[0].strip == 'PTP Hardware Clock'
+            result[parts[0].strip.tr(' ', '_')] = parts[1].strip
+          else
+            current_subsection[parts[0].strip.tr(' ', '_')] = parts[1].strip
+          end
         end
+      elsif current_subsection == result['Capabilities']
+        # Handle list items (capabilities and modes)
+        current_subsection[line.tr(' ', '_')] = true
+      elsif current_subsection == result['Hardware_Transmit_Timestamp_Modes'] ||
+            current_subsection == result['Hardware_Receive_Filter_Modes']
+        current_subsection[line.tr(' ', '_')] = true
       end
     end
 
     result
   end
 
   setcode do
+    physical_only = Facter.value(:ethtool_physical_only) || false
```

Comment:
Can you explain how this works? Are users expected to create a custom fact `ethtool_physical_only` and set it to `true`?

Reply:
Yes, only if they explicitly do not want virtual interfaces, the users create a custom fact `ethtool_physical_only` and set it to `true`. It is `false` by default to continue existing behaviour

I have added a little documentation to the `README` to explain how to do "Physical Interface Filtering"